components/sup/src/event.rs

```diff
@@ -106,12 +151,89 @@ pub fn publish(event: &impl Event) {
 #[derive(Clone, Debug)]
 pub struct EventCore {
     /// The unique identifier of the Supervisor sending the event.
-    pub supervisor_id: String,
+    supervisor_id: String,
+    ip_address: SocketAddr,
+    fqdn: String,
+    application: String,
+    environment: String,
+    meta: EventStreamMetadata,
+}
+
+impl EventCore {
+    pub fn new(config: &EventStreamConfig, sys: &Sys, fqdn: String) -> Self {
+        EventCore { supervisor_id: sys.member_id.clone(),
+                    ip_address: sys.gossip_listen(),
+                    fqdn,
+                    environment: config.environment.clone(),
+                    application: config.application.clone(),
+                    meta: config.meta.clone() }
+    }
+}
+
+/// Send an event for the start of a Service.
+pub fn service_started(service: &Service) {
+    if stream_initialized() {
+        publish(ServiceStartedEvent { service_metadata: Some(service.to_service_metadata()),
+                                      event_metadata:   None, });
+    }
+}
+
+/// Send an event for the stop of a Service.
+pub fn service_stopped(service: &Service) {
+    if stream_initialized() {
+        publish(ServiceStoppedEvent { service_metadata: Some(service.to_service_metadata()),
+                                      event_metadata:   None, });
+    }
+}
+
+/// `execution` will be `Some` if the service had a hook to run, and
+/// records how long it took that hook to execute completely.
+pub fn health_check(service: &Service, check_result: HealthCheck, execution: Option<Duration>) {
+    if stream_initialized() {
+        let check_result: types::HealthCheck = check_result.into();
+        publish(HealthCheckEvent { service_metadata: Some(service.to_service_metadata()),
+                                   event_metadata:   None,
+                                   result:           i32::from(check_result),
+                                   execution:        execution.map(Into::into), });
```

Comment:
I find this a bit more readable.
```suggestion
                                   execution:        execution.map(Duration::into), });
```
Also, I can't abide the redundancy of `Into::into` as long as we've gotta put something on the left side of the `::`, let's make it useful.